### DISCUSSION TITLE
API for products

### DISCUSSION THREAD

**Discussion by moskinov** [amzn/selling-partner-api-models#3732] [General]
I am a seller on Amazon and I have my own accounting system. How do I get my products from Amazon and upload products to Amazon via the api? Is there an open api - GET/POST/PATCH

**Comment by dev-pecaagora** [comment]
I wanted to know that too

**Comment by ritikacd** [comment]
@moskinov You can use [getListingsItem](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#getlistingsitem) API operation to get your listed products and [putListingsItem](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#putlistingsitem) API to create a new product listing on Amazon. 

For creating products in bulk, you may also use the [JSON_LISTINGS_FEED](https://developer-docs.amazon.com/sp-api/docs/listings-feed-schema-v2) feed type with the [Feeds API](https://developer-docs.amazon.com/sp-api/docs/feeds-api-v2021-06-30-reference).

Please refer to this guide that summarises the step-by-step process for creating and managing listings using SP-API https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#listing-items-on-amazon 

The guide talks about different workflows and APIs to integrate with to have a seamless experience with creating items on Amazon. 

Thanks,
Ritika Chaudhary
Selling Partner API Team

**Reply by yanyuk** [reply]
The `getListingsItem` call has proven to be unusable for us this past month. Our rate limit is nowhere near the documented 5 requests/second. We're experiencing successful calls on average once per minute, being throttled otherwise.

**Comment by dev-pecaagora** [comment]
How can I list all items without passing a sku?

**Comment by chapmanjw** [comment]
We are planning to launch a `searchListingsItems` operation on the Listings Items API in the future (that will work similarly to the `searchCatalogItems` operation on the Catalog Items API) which will allow you to paginate through listings for a given seller. However, in the meantime, you can use one of the inventory reports to get the listings for a given seller: https://developer-docs.amazon.com/sp-api/docs/report-type-values-inventory

Stay tuned for future launch announcements regarding the search operation for the Listings Items API.

**Reply by yanyuk** [reply]
@chapmanjw Is there any estimate for when this might be released? The single `getListingsItem` has inaccurate rate limit documentation (5 req/second). We're experiencing a rate limit of approximately 1 req/minute across ~2,000 customers. In testing with a seller ID that has not made this call for hours, we experience throttling on the first call. There seems to be no throttling based on developer/seller pair for this operation. Can you confirm?

Relevant case ID: 14331596151

**Comment by dev-pecaagora** [comment]
Response:

    "errors": [
        {
            "code": "Unauthorized",
            "message": "Access to requested resource is denied.",
            "details": ""
        }
    ]
}





Code:

// Access Token
        $access_token = "******";

        // Endpoint da API da América do Norte
        $endpoint = "https://sellingpartnerapi-na.amazon.com";

        // ID do Marketplace da Amazon dos EUA
        $marketplace_id = "A2Q3Y263D00KWC";

        // Parâmetros da solicitação para obter resumo do inventário
        $request_params = array(
            //"details" => true,  // Adicione os detalhes necessários
            "sellerId" => "****ZBF7IT2RR2", 
            "sku" => '342164',  // Use o marketplace_id como granularidadeId
            "marketplaceIds" => $marketplace_id
        );

        // Montando a URL da solicitação
        $url = $endpoint . "/listings/2021-08-01/items?" . http_build_query($request_params);


      
        // Configurando os cabeçalhos da solicitação
        $headers = array(
            "x-amz-access-token: " . $access_token,
            "Accept" => "application/json"
        );


       // $headers[] = 'Content-Type: application/json';

        // Inicializando a sessão cURL
        $ch = curl_init($url);

        // Definindo opções para a solicitação cURL
        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

        // Enviando a solicitação
        $response = curl_exec($ch);

        // Verificando por erros
        if (curl_errno($ch)) {
            echo 'Erro na solicitação cURL: ' . curl_error($ch);
        }

        // Fechando a sessão cURL
        curl_close($ch);

        // Imprimindo a resposta JSON formatada
        echo json_encode(json_decode($response), JSON_PRETTY_PRINT);

        die;

Help me :/

**Comment by dev-pecaagora** [comment]
![Captura de tela de 2023-11-14 10-12-55](https://github.com/amzn/selling-partner-api-docs/assets/56078526/b0f72340-dfaa-4c56-9fb5-ec0067ce218a)
